indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -16,55 +16,1991 @@
+            if (!verifyInitialRecordAndSkipExclusiveParition(record, initialOffsetsSnapshot)) {
+              continue;
+            }
+
+            if (log.isTraceEnabled()) {
+              log.trace(
+                  "Got stream[%s] partition[%s] sequence[%s].",
+                  record.getStream(),
+                  record.getPartitionId(),
+                  record.getSequenceNumber()
+              );
+            }
+
+            if (OrderedPartitionableRecord.END_OF_SHARD_MARKER.equals(record.getSequenceNumber())) {
+              // shard is closed
+              currOffsets.put(record.getPartitionId(), record.getSequenceNumber());
+            } else if (createSequenceNumber(record.getSequenceNumber()).compareTo(
+                createSequenceNumber(endOffsets.get(record.getPartitionId()))) <= 0) {
+
+
+              if (!record.getSequenceNumber().equals(currOffsets.get(record.getPartitionId()))) {
+                if (!ioConfig.isSkipOffsetGaps()) {
+                  throw new ISE(
+                      "WTF?! Got sequence[%s] after sequence[%s] in partition[%s].",
+                      record.getSequenceNumber(),
+                      currOffsets.get(record.getPartitionId()),
+                      record.getPartitionId()
+                  );
+                }
+              }
+
+              try {
+                final List<byte[]> valueBytess = record.getData();
+                final List<InputRow> rows;
+                if (valueBytess == null || valueBytess.isEmpty()) {
+                  rows = Utils.nullableListOf((InputRow) null);
+                } else {
+                  rows = new ArrayList<>();
+                  for (byte[] valueBytes : valueBytess) {
+                    rows.addAll(parser.parseBatch(ByteBuffer.wrap(valueBytes)));
+                  }
+                }
+                boolean isPersistRequired = false;
+
+                final SequenceMetadata sequenceToUse = sequences
+                    .stream()
+                    .filter(sequenceMetadata -> sequenceMetadata.canHandle(record))
+                    .findFirst()
+                    .orElse(null);
+
+                if (sequenceToUse == null) {
+                  throw new ISE(
+                      "WTH?! cannot find any valid sequence for record with partition [%d] and sequence [%d]. Current sequences: %s",
+                      record.getPartitionId(),
+                      record.getSequenceNumber(),
+                      sequences
+                  );
+                }
+
+                for (InputRow row : rows) {
+                  if (row != null && task.withinMinMaxRecordTime(row)) {
+                    final AppenderatorDriverAddResult addResult = driver.add(
+                        row,
+                        sequenceToUse.getSequenceName(),
+                        committerSupplier,
+                        // true for kafka, false for kinesis
+                        isSkipSegmentLineageCheck,
```

Comment:
Why is this false for Kinesis? If the Kafka implementation which combines multiple Kafka partitions into one Druid partition can't use the segment lineage check because of arbitrary ordering between Kafka partitions, I would expect Kinesis to have the same issue.

Reply:
yeah should be true for both kafka and kinesis